cdk/kraken/src/deploy.ts

```diff
@@ -25,68 +19,63 @@ export interface DeployJobProps {
-      image: 'pennlabs/helm-tools:39b60af248944898fcbc58d1fe5b0f1995420aef',
       defaultBranch: 'master',
       ...config,
     };
 
-    // Create job
     super(scope, 'deploy', {
       runsOn: 'ubuntu-latest',
-      container: {
-        image: fullConfig.image,
-      },
       if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
-      steps: [{
-        name: 'Deploy',
-        run: dedent`aws eks --region us-east-1 update-kubeconfig --name production --role-arn arn:aws:iam::\${AWS_ACCOUNT_ID}:role/kubectl
+      steps: [
+        {
+          id: 'synth',
+          name: 'Synth cdk8s manifests',
+          run: dedent`cd k8s
+          yarn install --frozen-lockfile
 
-        # get repo name (by removing owner/organization)
-        RELEASE_NAME=\${REPOSITORY#*/}
+          # get repo name (by removing owner/organization)
+          RELEASE_NAME=\${REPOSITORY#*/}
 
-        # this specifies what tag of icarus to pull down
-        DEPLOY_TAG=$(yq r k8s/values.yaml deploy_version)
-        if [ "$DEPLOY_TAG" = "null" ]; then
-            echo "Could not find deploy tag"
-            exit 1
-        fi
+          # Export RELEASE_NAME as an output
+          echo "::set-output RELEASE_NAME=$RELEASE_NAME"
+
+          yarn build`,
+          env: {
+            GIT_SHA: fullConfig.deployTag,
+            REPOSITORY: '${{ github.repository }}',
+            AWS_ACCOUNT_ID: '${{ secrets.AWS_ACCOUNT_ID }}',
+          },
+        },
+        {
+          name: 'Deploy',
+          run: dedent`aws eks --region us-east-1 update-kubeconfig --name production --role-arn arn:aws:iam::\${AWS_ACCOUNT_ID}:role/kubectl
 
-        helm repo add pennlabs https://helm.pennlabs.org/
-        for i in {1..10}; do
-          # This is bash soup, but it'll do.
-          # 1. Attempt to install with helm
-          # 2. If this succeeds, exit with a success status code
-          # 3. If it fails, mark the command as succeeded so that '-e' doesn't kick us out
-          # 4. Wait 10s and try again
-          helm upgrade --install --atomic --set=image_tag=$IMAGE_TAG -f k8s/values.yaml --version "\${DEPLOY_TAG}" $RELEASE_NAME pennlabs/icarus && exit 0 || true
-          sleep 10s
-          echo "Retrying deploy for $i times"
-        done
+          # get repo name from synth step
+          RELEASE_NAME=\${{ steps.synth.outputs.RELEASE_NAME }}
 
-        # If we get here, all helm installs failed so our command should fail
-        exit 1`,
-        env: {
-          IMAGE_TAG: fullConfig.deployTag,
-          AWS_ACCOUNT_ID: '${{ secrets.AWS_ACCOUNT_ID }}',
-          AWS_ACCESS_KEY_ID: '${{ secrets.GH_AWS_ACCESS_KEY_ID }}',
-          AWS_SECRET_ACCESS_KEY: '${{ secrets.GH_AWS_SECRET_ACCESS_KEY }}',
-          DO_AUTH_TOKEN: '${{ secrets.DO_AUTH_TOKEN }}',
-          REPOSITORY: '${{ github.repository }}',
+          # Deploy
+          kubectl apply -f k8s/dist/ -l app.kubernetes.io/component=certificate
```

Comment:
Do we ever foresee in the future generating other files in the `k8s/dist/` folder? Because if so, we may want to specify the file more explicitly. 

However, I thought about this and feel like we won't (if it's generated in `k8s/dist/` and it's not deployment yaml, then it doesn't belong there). If you agree, then this looks good to me.

Reply:
I don't, but I actually like the idea of having this `dist` folder where each product can just add any kube manifests they want deployed

Reply:
Wait a second, are we encouraging developers to write yaml now :P

Reply:
Well, ideally they wouldn't have to touch any yaml, but having this folder exist as a breakglass option is nice